sc-poc/sources/auth/opapolicy/source.go

```diff
@@ -0,0 +1,114 @@
+package opapolicy
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/spacecloud-io/space-cloud/managers/source"
+	"github.com/spacecloud-io/space-cloud/modules/auth"
+	authTypes "github.com/spacecloud-io/space-cloud/modules/auth/types"
+	"github.com/spacecloud-io/space-cloud/modules/graphql"
+	"github.com/spacecloud-io/space-cloud/pkg/apis/core/v1alpha1"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/mitchellh/mapstructure"
+	"github.com/open-policy-agent/opa/rego"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var opapolicyResource = schema.GroupVersionResource{Group: "core.space-cloud.io", Version: "v1alpha1", Resource: "opapolicies"}
+
+func init() {
+	source.RegisterSource(OPAPolicySource{}, opapolicyResource)
+}
+
+// OPAPolicySource describes the OPAPolicy source
+type OPAPolicySource struct {
+	v1alpha1.OPAPolicy
+
+	// For internal use
+	logger                 *zap.Logger
+	preparedQuery          rego.PreparedEvalQuery
+	compiledGraphQLQueries map[string]*graphql.CompiledQuery
+}
+
+// CaddyModule returns the Caddy module information.
+func (OPAPolicySource) CaddyModule() caddy.ModuleInfo {
+	return caddy.ModuleInfo{
+		ID:  caddy.ModuleID(source.GetModuleName(opapolicyResource)),
+		New: func() caddy.Module { return new(OPAPolicySource) },
+	}
+}
+
+// Provision provisions the source
+func (s *OPAPolicySource) Provision(ctx caddy.Context) error {
+	s.logger = ctx.Logger(s)
+
+	// Compile the rego policies
+	if err := s.compileRegoPolicy(); err != nil {
```

Comment:
You can't call this here since you map will most likely be empty at this point. You should call this within `SetConpiledQueries` function.